extension/lib/garbage-collector.js

```diff
@@ -5,68 +5,25 @@
 "use strict";
 
 const { Cc, Ci, Cu } = require("chrome");
-const { EventEmitter } = require("sdk/deprecated/events");
+const { emit, on, off } = require("sdk/event/core");
 const prefs = require("sdk/preferences/service");
 const timers = require("sdk/timers");
 const unload = require("sdk/system/unload");
 
 const config = require("./config");
 
-Cu.import('resource://gre/modules/Services.jsm');
-
-const reporter = EventEmitter.compose({
-  _pref_gc_notifications: null,
-
-  constructor: function Reporter() {
-    // Report unhandled errors from listeners
-    this.on("error", console.exception.bind(console));
-
-    // Make sure we clean-up correctly
-    unload.ensure(this, 'unload');
-
-    if (config.application.branch >= 16) {
-      this._pref_gc_notifications = config.preferences.memory_notify;
-    }
-    else {
-      this._pref_gc_notifications = config.preferences.memory_log;
-    }
+const PREF_GC_NOTIFICATIONS = config.preferences.memory_notify;
 
-    // Ensure GC/CC observer and console messages preference is enabled
-    this._isEnabled = prefs.get(this._pref_gc_notifications);
-    if (!this._isEnabled)
-      this._enable();
-
-    Services.obs.addObserver(this, config.application.topic_cc_statistics, false);
-    Services.obs.addObserver(this, config.application.topic_gc_statistics, false);
-  },
-
-  get pref_gc_notifications() {
-    return this._pref_gc_notifications;
-  },
-
-  unload: function Reporter_unload() {
-    this._removeAllListeners();
-
-    Services.obs.removeObserver(this, config.application.topic_cc_statistics, false);
-    Services.obs.removeObserver(this, config.application.topic_gc_statistics, false);
-  },
-
-  _enable: function Reporter__enable() {
-    var modifiedPrefs = JSON.parse(prefs.get(config.preferences.modified_prefs,
-                                             "{}"));
-    if (!modifiedPrefs.hasOwnProperty(this._pref_gc_notifications)) {
-      modifiedPrefs[this._pref_gc_notifications] = prefs.get(this._pref_gc_notifications);
-    }
+Cu.import('resource://gre/modules/Services.jsm');
```

Comment:
Maybe then leave it. If it is worth the time we can have a refactoring later. Right now we simply need the code working, and that's really tiny nit which doesnt block us.